app/src/main/kotlin/com/aliucord/manager/ui/viewmodel/UpdaterViewModel.kt

```diff
@@ -0,0 +1,91 @@
+package com.aliucord.manager.ui.viewmodel
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.aliucord.manager.BuildConfig
+import com.aliucord.manager.domain.manager.DownloadManager
+import com.aliucord.manager.domain.repository.GithubRepository
+import com.aliucord.manager.installer.util.installApks
+import com.aliucord.manager.network.utils.SemVer
+import com.aliucord.manager.network.utils.getOrNull
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class UpdaterViewModel(
+    private val github: GithubRepository,
+    private val downloadManager: DownloadManager,
+    private val application: Application
+) : ViewModel() {
+    var showDialog by mutableStateOf(false)
+        private set
+    var targetVersion by mutableStateOf<String?>(null)
+        private set
+    var isWorking by mutableStateOf(false)
+        private set
+
+    private var targetApkUrl: String? = null
+
+    fun dismissDialog() {
+        showDialog = false
+    }
+
+    fun triggerUpdate() {
+        if (isWorking) return
+        val url = targetApkUrl ?: return
+
+        viewModelScope.launch {
+            isWorking = true
+
+            application.externalCacheDir!!.resolve("manager.apk").apply {
+                delete()
+
+                downloadManager.download(url, this)
+                application.installApks(this)
+
+                delay(2000)
+                showDialog = false
+
+                delay(10000)
```

Comment:
why these delays

Reply:
The first one was for human psycology (if window closes immediately after prompting, then "installs done guh" but it could have not even prompted the install window yet)

2nd one doesnt have a purpose anymore will remove

Reply:
you spelt psychology wrong

Reply:
death

Reply:
you used death in the wrong sense

Reply:
explode

Reply:
Missing capital letters at the beginning of sentences, and periods at the ends. The 'P' in Psychology is not capitalized. "death" is not a complete sentence.

Reply:
Missing "bitches"

Reply:
Missing "b"

Reply:
not enough uwus